lib/buffer.js

```diff
@@ -241,6 +241,8 @@ Buffer.concat = function(list, length) {
   var pos = 0;
   for (let i = 0; i < list.length; i++) {
     var buf = list[i];
+    if (!(buf instanceof Buffer))
```

Comment:
I was suggesting that because, if we change how we detect Buffers (which will be implemented in `isBuffer`) later on, then we don't have to change this line.